packages/errors/src/__tests__/json-rpc-error.ts

```diff
@@ -0,0 +1,52 @@
+import { SOLANA_ERROR__JSON_RPC__SERVER_ERROR_SEND_TRANSACTION_PREFLIGHT_FAILURE, SolanaErrorCode } from '../codes';
+import { SolanaErrorContext } from '../context';
+import { SolanaError } from '../error';
+import { getSolanaErrorFromJsonRpcError } from '../json-rpc-error';
+import { getSolanaErrorFromTransactionError } from '../transaction-error';
+
+jest.mock('../transaction-error.ts');
+
+describe('getSolanaErrorFromJsonRpcError', () => {
+    it('produces a `SolanaError` given a JSON-RPC error with data', () => {
+        const expectedData = Symbol() as unknown as SolanaErrorContext[SolanaErrorCode];
+        const error = getSolanaErrorFromJsonRpcError({ code: 123, data: expectedData, message: 'o no' });
+        expect(error).toEqual(new SolanaError(123 as SolanaErrorCode, expectedData));
```

Comment:
I am sliiiiighly concerned about using RPC error codes as Solana error codes directly. Right now they are negative and cannot conflict with the rest of our codes but what if this change with RPC v2?

Reply:
They're ‘[reserved](https://www.jsonrpc.org/specification#error_object)’ by the JSON-RPC 2.0 specification, but I can always introduce an offset so that they're positive and unique within our `SolanaErrorCode` universe.

Reply:
I made this change (ie. added an offset) and I think it made things way worse.

Reply:
No I think it's best without the offset. I was just saying we're merging two error code system together and right now it's okay because they don't overlap but are we worried that they might in the future?